lib/puck/dependency_resolver.rb

```diff
@@ -63,7 +63,12 @@ def contained_bundler(gem_home, gemfile, lockfile, groups)
           EOS
           result, error, message, backtrace = Marshal.load(String.from_java_bytes(unit.run))
           if error
-            raise error, message, Array(backtrace)+caller
+            arity = error.method(:new).arity
+            if arity > 1 || arity < 0
+              raise RuntimeError, error.name + ': ' + message, Array(backtrace)+caller
```

Comment:
Opted for `'%s: %s' % [error.name, message]` as that's the style that's used on line 19 in this file.